scylla-cql/src/types/serialize/row.rs

```diff
@@ -126,13 +132,156 @@ pub enum ValueListToSerializeRowAdapterError {
     NoBindMarkerWithName { name: String },
 }
 
+/// A buffer containing already serialized values.
+///
+/// It is not aware of the types of contained values,
+/// it is basically a byte buffer in the format expected by the CQL protocol.
+/// Usually there is no need for a user of a driver to use this struct, it is mostly internal.
+/// The exception are APIs like `ClusterData::compute_token` / `ClusterData::get_endpoints`.
+/// Allows adding new values to the buffer and iterating over the content.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct NewSerializedValues {
```

Comment:
`NewSerializedValues` will not be "new" forever, I think it makes sense to just use `SerializedValues` here and rename the old one to `LegacySerializedValues` and eventually remove it.